nautobot/extras/views.py

```diff
@@ -455,6 +459,109 @@ class CustomFieldBulkDeleteView(generic.BulkDeleteView):
     queryset = CustomField.objects.all()
     table = tables.CustomFieldTable
 
+    def construct_custom_field_delete_tasks(self, queryset):
+        """
+        Helper method to construct a list of celery tasks to execute when bulk deleting custom fields.
+        """
+        # si (*args, **kwargs) creates immuatable signature.
+        # That makes the result of the task is independent of that of the previous task
+        tasks = [
+            delete_custom_field_data.si(obj.name, set(obj.content_types.values_list("pk", flat=True)))
+            for obj in queryset
+        ]
+        return tasks
+
+    # Overloading post() method for CustomFieldBulkDeleteView
```

Comment:
Instead of having to copy-and-paste the entire function from the base class, can we add an extension point to the base class instead?

Something like:

```python
class BulkDeleteView(GetReturnURLMixin, ObjectPermissionRequiredMixin, View):
    ...
    def post(self, request, **kwargs):
        ...
        if form.is_valid():
                logger.debug("Form validation was successful")

                # Delete objects
                queryset = self.queryset.filter(pk__in=pk_list)

                self.perform_pre_delete(queryset)  # <-----<< added
                ...

        def perform_pre_delete(self):
            pass
```

```python
class CustomFieldBulkDeleteView(generic.BulkDeleteView):
    ...
    def perform_pre_delete(self, queryset):
        tasks = self.construct_custom_field_delete_tasks(queryset=queryset)
        ...
```